File: src/com/esielkar/bridge/BridgeExample.java
package com.esielkar.bridge;

public class BridgeExample {
    public static void main(String[] args) {
        Platform platform = new ConcretePlatform();
        Abstraction abstraction = new Abstraction(platform);
        RefinedAbstraction refinedAbstraction = new RefinedAbstraction(platform);

        abstraction.feature1();
        abstraction.feature2();

        refinedAbstraction.featureN();

    }
}

File: dart/design_patterns/bin/bridge/bridge_example.dart
import 'abstraction.dart';
import 'platform.dart';

void main(List<String> args) {
  Platform platform = ConcretePlatform();
  Abstraction abstraction = Abstraction(platform);
  RefinedAbstraction refinedAbstraction = RefinedAbstraction(platform);

  abstraction.feature1();
  abstraction.feature2();

  refinedAbstraction.featureN();
}
